File: scaffold/src/main/java/top/xuqingquan/web/system/MiddlewareWebChromeBase.java
package top.xuqingquan.web.system;

import android.webkit.WebChromeClient;

public class MiddlewareWebChromeBase extends WebChromeClientDelegate {

    private MiddlewareWebChromeBase mMiddlewareWebChromeBase;

    @SuppressWarnings("WeakerAccess")
    public MiddlewareWebChromeBase(WebChromeClient webChromeClient) {
        super(webChromeClient);
    }

    @SuppressWarnings("WeakerAccess")
    public MiddlewareWebChromeBase() {
        super(null);
    }

    @Override
    public final void setDelegate(WebChromeClient delegate) {
        super.setDelegate(delegate);
    }

    @SuppressWarnings("UnusedReturnValue")
    public final MiddlewareWebChromeBase enq(MiddlewareWebChromeBase middlewareWebChromeBase) {
        setDelegate(middlewareWebChromeBase);
        this.mMiddlewareWebChromeBase = middlewareWebChromeBase;
        return this.mMiddlewareWebChromeBase;
    }


    public final MiddlewareWebChromeBase next() {
        return this.mMiddlewareWebChromeBase;
    }

}

File: scaffold/src/main/java/top/xuqingquan/webbak/system/MiddlewareWebChromeBase.kt
package top.xuqingquan.webbak.system

import android.webkit.WebChromeClient

open class MiddlewareWebChromeBase(webChromeClient: WebChromeClient? = null) :
    WebChromeClientDelegate(webChromeClient) {

    private var mMiddlewareWebChromeBase: MiddlewareWebChromeBase? = null

    override var delegate: WebChromeClient?
        get() = super.delegate
        set(delegate) {
            super.delegate = delegate
        }

    fun enq(middlewareWebChromeBase: MiddlewareWebChromeBase): MiddlewareWebChromeBase? {
        delegate = middlewareWebChromeBase
        this.mMiddlewareWebChromeBase = middlewareWebChromeBase
        return this.mMiddlewareWebChromeBase
    }


    operator fun next(): MiddlewareWebChromeBase? {
        return this.mMiddlewareWebChromeBase
    }

}
